src/System.Memory/src/System/Span.cs

```diff
@@ -346,6 +328,12 @@ public override int GetHashCode()
             throw new NotSupportedException(SR.CannotCallGetHashCodeOnSpan);
         }
 
+        /// <summary>
+        /// Returns a <see cref="String"/> with the name of the type and the number of elements
+        /// </summary>
+        /// <returns>A <see cref="String"/> with the name of the type and the number of elements</returns>
+        public override string ToString() => $"System.Span[{Length}]";
```

Comment:
Should this  be identical to the DebuggerDisplay string (i.e. should we include typeof(T).Name to ToString) - [SpanDebuggerDisplay](https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/Span.cs#L128) / [ReadOnlySpanDebuggerDisplay](https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/ReadOnlySpan.cs#L124)?

Similarly, should we add the prefix "System.Span" to the DebuggerDisplay, with angle brackets around the type T?

For instance, I propose the following:
```C#
Span<int> span = new int[10];
string str = span.ToString(); //System.Span<int>[10]
```

Reply:
Sure, that would be useful.

Reply:
Updated with the new ToString/DebuggerDisplay string.

I'll post a PR to coreclr with the new strings to FastSpan along with the removal of the old NonPortableCast functions after this PR in corefx is merged.